src/OpenTK.Windowing.GraphicsLibraryFramework/JoystickState.cs

```diff
@@ -0,0 +1,197 @@
+//
+// JoystickState.cs
+//
+// Copyright (C) OpenTK
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+using OpenTK.Windowing.Common.Input;
+
+[assembly: InternalsVisibleTo("OpenTK.Windowing.Desktop")]
+
+namespace OpenTK.Windowing.GraphicsLibraryFramework
+{
+    /// <inheritdoc />
+    internal sealed class JoystickState : IJoystickState
+    {
+        private const int StatePositionCount = 2;
+
+        private int _current = StatePositionCount - 1;
+        private int _previous = StatePositionCount - 2;
+
+        private Hat[][] _hats;
+        private float[][] _axes;
+        private byte[][] _buttons;
```

Comment:
The `_name` naming convention is usually for backing fields of properties. I wouldn't consider these properties.

Reply:
In my own code I hate properties _and_ I hate underscores preceding field names. In this codebase, StyleCop demands that all fields are prefixed with underscores.

Reply:
Ah that would be all private fields I think, because I've never had that problem with stylecop.